### ISSUE TITLE
Web UI: consistent pagination for clients and servers

### ISSUE THREAD

**Issue by tjohnston-cd** [hashicorp/nomad#18083]
### Proposal
Currently using 1.5.6 (but I suspect this is the same in newer versions, based on the release changelogs):

The `clients` webui page has a "per-page" dropdown box to set how many should be shown:
 
![perpage-clients](https://github.com/hashicorp/nomad/assets/42251505/4a09ef0b-694f-4f3e-9832-de1a750bb188)

The `servers` webui page does not have this. Instead, it seems to choose its pagination differently.

For example, in my case, I have to navigate to page 2 to see my 9th server's details.

![servers](https://github.com/hashicorp/nomad/assets/42251505/2eae498e-fb08-4f3c-ae79-933e0803614e)

It's a minor difference, but if both behaved like the clients page.. it would be a bit easier to use - and more consistent!

### Use-cases
I would use the webUI the same way, but it would be a bit easier to navigate / set pagination the way I want on these panels.

Maybe fixing this up leads to some refactoring/cleanup behind the scenes..? Not sure..

### Attempted Solutions
N/A

**Comment by philrenaud** [comment]
Good call, @tjohnston-cd — we'll whip something up to fix this. Thanks for the heads up.
